File: src/java/sorts/BubbleSort.java
package java.sorts;

/**
 * @fileName: BubbleSort.java
 * @author: wyli
 * @createTime: 2023/9/3 21:45
 * @description:
 */
public class BubbleSort {

    public static void bubbleSort(int[] a, int n){
        if (n <= 1) return;
        boolean flag = false;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (a[j] > a[j + 1]){
                    int tmp = a[j];
                    a[j] = a[j + 1];
                    a[j + 1] = tmp;
                    flag = true;
                }
            }
            if (!flag) break;
        }
    }

}

File: src/cpp/sorts/BubbleSort.cpp
#include<iostream>
#include<vector>

class BubbleSort{
public:
    void bubbleSort(std::vector<int>& arr){
        int len = arr.size();
        if(len <= 1){
            return;
        }
        bool flag = false;
        for (int i = 0; i < len; i++)
        {
            for (int j = 0; i < len - i - 1; i++)
            {
                if(arr[j] > arr[j + 1]){
                    int tmp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = tmp;
                    flag = true;
                }
            }
            if (!flag)
            {
                break;
            }
            
        }
        
    }
};
